test/gerkin/features/data_update_or_create.feature

```diff
@@ -1,22 +1,19 @@
 Feature: API endpoint allowing users to create a new record if it not exists in the Digital Registries database or update records if already exists.
   Request endpoint: POST /data/{registryname}/{versionnumber}/update-or-create
 
-  Background:
-    Given The user wants to create a new record or update an existing one if already exists in the Digital Registries database
-
   Scenario: The user successfully creates new record in the Digital Registries database
-    Given The requested record does not exist in the database
-    When The user triggers an action to create a new record in the database
-    And The request with a valid payload is sent
-    Then Operation finishes successfully
+    Given The user wants to create a new record "Jon Snake" or update an existing one if already exists in the Digital Registries database
+    And The requested record "Jon Snake" does not exist in the database
+    When The user triggers an action to create record "Jon Snake" in the database
+    Then Operation to create "Jon Snake" finishes successfully
 
   Scenario: The user successfully updates the already existing record in the Digital Registries database
-    Given The requested record already exist in the database
-    When The user triggers an action to create a new record in the database
-    And The request with a valid payload is sent
-    Then Operation finishes successfully
+    Given The user wants to create a new record "Johny Small" or update an existing one if already exists in the Digital Registries database
+    And The requested record "Johny Small" already exists in the database
+    When The user triggers an action to update record "Johny Small" in the database
+    Then Operation to update record "Johny Small" finishes successfully
 
   Scenario: The user is not able to create/update a record in the Digital Registries database
-    When The user triggers an action to create a new record in the database
-    And The request with an invalid payload is sent
-    Then Operation results in an error
+    Given Given The user wants to create record "Ali Smith" or update an existing one if already exists in the Digital Registries database
+    When The user triggers an action to create record "Ali Smith" in the database
+    Then Operation results to create/update record "Ali Smith" in an error
```

Comment:
Should we add any information of why create/update will fail? E.g. invalid payload/missing header/not authorized?

Reply:
This information is in the title of the scenario. We can discuss it in the stand-up meeting because most scenarios are written like this :)